src/types.rs

```diff
@@ -335,10 +337,66 @@ pub struct SignedMessage {
 
 pub type RpcResult<T> = Result<T, RpcError>;
 
+#[derive(Clone, Debug, CandidType, Deserialize)]
+pub enum MultiRpcResult<T> {
+    Consistent(RpcResult<T>),
+    Inconsistent(Vec<(RpcService, RpcResult<T>)>),
+}
+
+impl<T> MultiRpcResult<T> {
+    pub fn map<R>(self, f: impl Fn(T) -> R) -> MultiRpcResult<R> {
```

Comment:
nit: I would add unit tests to ensure the that function is executed on all ok variants.